docs/source/examples/mg-mode-part-i.ipynb

```diff
@@ -0,0 +1,707 @@
+    "\n",
+    "# The scattering matrix is ordered with incoming groups as rows and outgoing groups as columns\n",
+    "# (i.e., above the diagonal is up-scattering).\n",
+    "scatter_matrix = \\\n",
+    "    [[[1.27537E-1, 4.23780E-2, 9.43740E-6, 5.51630E-9, 0.00000E-0, 0.00000E-0, 0.00000E-0],\n",
+    "      [0.00000E-0, 3.24456E-1, 1.63140E-3, 3.14270E-9, 0.00000E-0, 0.00000E-0, 0.00000E-0],\n",
+    "      [0.00000E-0, 0.00000E-0, 4.50940E-1, 2.67920E-3, 0.00000E-0, 0.00000E-0, 0.00000E-0],\n",
+    "      [0.00000E-0, 0.00000E-0, 0.00000E-0, 4.52565E-1, 5.56640E-3, 0.00000E-0, 0.00000E-0],\n",
+    "      [0.00000E-0, 0.00000E-0, 0.00000E-0, 1.25250E-4, 2.71401E-1, 1.02550E-2, 1.00210E-8],\n",
+    "      [0.00000E-0, 0.00000E-0, 0.00000E-0, 0.00000E-0, 1.29680E-3, 2.65802E-1, 1.68090E-2],\n",
+    "      [0.00000E-0, 0.00000E-0, 0.00000E-0, 0.00000E-0, 0.00000E-0, 8.54580E-3, 2.73080E-1]]]\n",
+    "scatter_matrix = np.array(scatter_matrix)\n",
+    "scatter_matrix = np.rollaxis(scatter_matrix, 0, 3)\n",
+    "uo2_xsdata.set_scatter_matrix(scatter_matrix, temperature=294.)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now that the UO2 data has been created, we can move on to the remaining materials using the same process.\n",
+    "\n",
+    "However, we will actually skip repeating the above for now.  Our simulation will instead use the `c5g7.h5` file that has already been created using exactly the same logic as above, but for the remaining materials in the benchmark problem.\n",
+    "\n",
+    "For now we will show how you would use the `uo2_xsdata` information to create an `openmc.MGXSLibrary` object and write to disk."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# Initialize the library\n",
+    "mg_cross_sections_file = openmc.MGXSLibrary(groups)\n",
+    "\n",
+    "# Add the UO2 data to it\n",
+    "mg_cross_sections_file.add_xsdata(uo2_xsdata)\n",
+    "\n",
+    "# And write to disk\n",
+    "mg_cross_sections_file.export_to_hdf5('mgxs.h5')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Generate 2-D C5G7 Problem Input Files"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To build the actual 2-D model, we will first begin by creating the `materials.xml` file.\n",
+    "\n",
+    "First we need to define materials that will be used in the problem. In other notebooks, either `openmc.Nuclides` or `openmc.Elements` were created at the equivalent stage. We can do that in multi-group mode as well.  However, multi-group cross-sections are sometimes provided as macroscopic cross-sections; the C5G7 benchmark data are macroscopic. In this case, we can instead use `openmc.Macroscopic` objects to in-place of `openmc.Nuclides` or `openmc.Elements`.\n",
```

Comment:
`openmc.Nuclide` and `openmc.Element` .. if you really want to make it plural, put the -s outside of backticks.